SpokestackTests/TFLiteWakewordRecognizerTest.swift

```diff
@@ -0,0 +1,234 @@
+}
+
+class TFLiteWakewordRecognizerTestDelegate: PipelineDelegate, SpeechEventListener {
+    /// Spy pattern for the system under test.
+    /// asyncExpectation lets the caller's test know when the delegate has been called.
+    var didError: Bool = false
+    var deactivated: Bool = false
+    var activated: Bool = false
+    var asyncExpectation: XCTestExpectation?
+    var didActivateExpectation: XCTestExpectation?
+
+    func reset() {
+        self.didError = false
+        self.deactivated = false
+        self.activated = false
+        asyncExpectation = .none
+        self.didActivateExpectation = .none
+    }
+    
+    func didRecognize(_ result: SpeechContext) {}
+    
+    func failure(speechError: Error) {
+        print(speechError)
+        guard let _ = asyncExpectation else {
```

Comment:
Seems like what you want here is just an if branch that fails with `speechError` if no expectation is set and fulfills the expectation (with no `?` needed) if it is set—same for `didActivate`.

Reply:
I thought this too, but that approach fails for an unknown reason in multi-threaded tests (not necessarily here, but in other tests), so I standardized on this approach.

Reply:
Well that's...disconcerting.